src/librbd/io/ImageRequestWQ.cc

```diff
@@ -597,6 +597,12 @@ void *ImageRequestWQ<I>::_void_dequeue() {
     return nullptr;
   }
 
+  if (write_op) {
+    m_image_ctx.write_iops_throttle->get(1);
```

Comment:
If I make this async-ed, I believe the problem you mentioned will be gone, right?